client/dashboard/me/profile-personal-details/update-email/email-verification-banner.tsx

```diff
@@ -0,0 +1,185 @@
+import { resendEmailVerificationMutation } from '@automattic/api-queries';
+import { useMutation } from '@tanstack/react-query';
+import { Link } from '@tanstack/react-router';
+import { Button } from '@wordpress/components';
+import { useDispatch } from '@wordpress/data';
+import { createInterpolateElement } from '@wordpress/element';
+import { __, sprintf } from '@wordpress/i18n';
+import { store as noticesStore } from '@wordpress/notices';
+import { useState, useEffect } from 'react';
+import Notice from '../../../components/notice';
+import type { UserSettings } from '@automattic/api-core';
+
+// Get email verification params from URL
+function getEmailVerificationParams() {
+	if ( typeof window === 'undefined' ) {
```

Comment:
@pandah3 I'm mostly curious: but is this for the tests or have you faced the error of window being undefined?